src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/TrieRangeIterator.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v1.sortedterms;
+
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.google.common.collect.AbstractIterator;
+
+import org.apache.cassandra.index.sai.disk.v1.trie.TrieTermsDictionaryReader;
+import org.apache.cassandra.io.tries.Walker;
+import org.apache.cassandra.io.util.Rebufferer;
+import org.apache.cassandra.utils.Pair;
+import org.apache.cassandra.utils.SizedInts;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+
+/**
+ * Thread-unsafe term -> long trie range iterator.
+ * The payload may be changed to something else.
+ */
+@NotThreadSafe
+class TrieRangeIterator extends Walker<TrieRangeIterator>
+{
+    private final ByteSource limit;
+    private final TrieTermsDictionaryReader.TransitionBytesCollector collector = new TrieTermsDictionaryReader.TransitionBytesCollector();
```

Comment:
nit: Depending on which way we go w/ trying to keep the stuff in the `sortedterms` package SAI-agnostic, a `Collector` interface might make sense here (and passing the collector into the constructor, etc.)

Reply:
Class removed